pkg/provider/provider.go

```diff
@@ -541,6 +549,13 @@ func (k *kubeProvider) Read(ctx context.Context, req *pulumirpc.ReadRequest) (*p
 	if readErr != nil {
 		glog.V(3).Infof("%v", readErr)
 
+		if meta.IsNoMatchError(readErr) {
+			// If it's a "no match" error, this is probably a CustomResource with no corresponding
+			// CustomResourceDefinition. This usually happens if the CRD was deleted, and it's safe
+			// to consider the CR to be deleted as well in this case.
+			return &pulumirpc.ReadResponse{Id: "", Properties: nil}, nil
```

Comment:
Can we move this "deleting `ReadResponse`" into a variable? We're doing this other places and it's a weird, bespoke thing to dup everywhere, it seems to me.